controllers/openstackmachine_controller.go

```diff
@@ -422,7 +422,7 @@ func (r *OpenStackMachineReconciler) reconcileMachineState(scope *scope.WithLogg
 		conditions.MarkTrue(openStackMachine, infrav1.InstanceReadyCondition)
 
 		// Set properties required by CAPI machine controller
-		openStackMachine.Spec.ProviderID = ptr.To(fmt.Sprintf("openstack://%s/%s", openStackMachine.Spec.IdentityRef.Region, *openStackServer.Status.InstanceID))
+		openStackMachine.Spec.ProviderID = ptr.To(fmt.Sprintf("openstack://%s/%s", (*openStackMachine.Spec.IdentityRef).Region, *openStackServer.Status.InstanceID))
```

Comment:
That's still going to panic when IdentityRef is nil. How about something like:

```go
var region string
if openStackMachine.Spec.IdentityRef != nil {
    region = openStackMachine.Spec.IdentityRef.Region
}
openStackMachine.Spec.ProviderID = ptr.To(fmt.Sprintf("openstack://%s/%s", region, *openStackServer.Status.InstanceID))
```

Reply:
Oh, IdentityRef could be nil I did'nt get It !

I'll modify that Thx.